Tests/WorkspaceTests/WorkspaceStateTests.swift

```diff
@@ -0,0 +1,80 @@
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import TSCBasic
+@testable import Workspace
+import XCTest
+
+final class WorkspaceStateTests: XCTestCase {
+    func testSavedDependenciesAreSorted() throws {
+        let fs = InMemoryFileSystem()
+
+        let buildDir = AbsolutePath("/.build")
+        let statePath = buildDir.appending(component: "workspace-state.json")
+
+        try fs.writeFileContents(statePath) {
+            $0 <<<
+                """
+                {
+                    "object": {
+                        "artifacts": [],
+                        "dependencies": [
+                            {
+                                "basedOn": null,
+                                "packageRef": {
+                                  "identity": "yams",
+                                  "kind": "remoteSourceControl",
+                                  "location": "https://github.com/jpsim/Yams.git",
+                                  "name": "Yams"
+                                },
+                                "state": {
+                                  "checkoutState": {
+                                    "revision": "9ff1cc9327586db4e0c8f46f064b6a82ec1566fa",
+                                    "version": "4.0.6"
+                                  },
+                                  "name": "checkout"
+                                },
+                                "subpath": "Yams"
+                            },
+                            {
+                                "basedOn": null,
+                                "packageRef": {
+                                  "identity": "swift-argument-parser",
+                                  "kind": "remoteSourceControl",
+                                  "location": "https://github.com/apple/swift-argument-parser.git",
+                                  "name": "swift-argument-parser"
+                                },
+                                "state": {
+                                  "checkoutState": {
+                                    "revision": "83b23d940471b313427da226196661856f6ba3e0",
+                                    "version": "0.4.4"
+                                  },
+                                  "name": "checkout"
+                                },
+                                "subpath": "swift-argument-parser"
+                            }
+                        ]
+                    },
+                    "version": 4
+                }
+                """
+        }
+
+        let state = WorkspaceState(dataPath: buildDir, fileSystem: fs)
+        try state.save()
+
+        let serialized = try fs.readFileContents(statePath).description
+
+        let argpRange = try XCTUnwrap(serialized.range(of: "swift-argument-parser"))
+        let yamsRange = try XCTUnwrap(serialized.range(of: "yams"))
+
+        XCTAssertTrue(argpRange.lowerBound < yamsRange.lowerBound)
```

Comment:
IIRC there is hash randomization that potentially causes the hash order to be different for each run, so in theory this should pass 50% of the time or fail 50% of the time even without sorting, which is better than accidentally passing all the time.  A potentially more consistent way of doing it might be programmatically generating the input with, say, 100 unique input strings, and then checking that the sorted set of unique appear in increasing order.  I don't think it has to be done for this PR since we'll still get a signal from sporadic failures, just a thought for a possible improvement in the future.